lib/kitchen/transport/base.rb

```diff
@@ -138,9 +138,12 @@ def execute_with_retry(command, retryable_exit_codes = [], max_retries = 1, wait
         end
 
         def retry?(current_try, max_retries, retryable_exit_codes, exit_code)
-          current_try <= max_retries &&
-            !retryable_exit_codes.nil? &&
-            retryable_exit_codes.include?(exit_code)
+          if !retryable_exit_codes.nil? && retryable_exit_codes.class == Array || retryable_exit_codes.class == Integer
```

Comment:
Explicitly checking classes like this leads to fragile code. To `to_i` will already convert things like `"1"` so if the concern is string like `"1,2"` we should either check for that explicitly or just support it.

```ruby
Array(retryable_exit_codes).flatten.flat_map{|v| v.is_a?(String) ? v.split(/,/) : v}.map(&:to_i).include?(exit_code)
```

Reply:
Hi @coderanger, If we send values in kitchen.yml file without space as 
```
retry_on_exit_code: 
   -35
   -1 
```
then we get value in string format with negative integer as `"-35 -1"`.

And as per YML documentation `sequence use a dash followed by a space` : https://symfony.com/doc/current/components/yaml/yaml_format.html#collections.

So if we make string validation with ` Array(retryable_exit_codes).flatten.flat_map{|v| v.is_a?(String) ? v.split(/,/) : v}.map(&:to_i).include?(exit_code) ` then we are allowing the user to pass `invalid yml format` .

Please let me know if we should allow the invalid format or raise error in that case?